src/comgr.cpp

```diff
@@ -115,14 +123,71 @@ static void AddOcl20CompilerOptions(OptionList& list)
 /// (or even can be harmful) for building via comgr layer.
 ///
 /// \todo Produce proper options in, er, proper places, and get rid of this.
-static void RemoveSuperfluousOptions(OptionList& list)
+static void RemoveOclOptionsUnwanted(OptionList& list)
 {
     list.erase(remove_if(list.begin(),
                          list.end(),
                          [&](const auto& option) { return StartsWith(option, "-mcpu="); }),
                list.end());
 }
 
+static auto GetOptionsNoSplit()
+{
+    static const std::vector<std::string> rv = {
+        "-isystem", "-L", "-Wl,-rpath", "-Xclang", "-hip-path", "-mllvm", "-x"};
+    return rv;
+}
+
+namespace hip {
+
+static bool IsLinkerOption(const std::string& option)
+{
+    return miopen::StartsWith(option, "-L") || miopen::StartsWith(option, "-Wl,") ||
+           option == "-ldl" || option == "-lm" || option == "--hip-link";
+}
+
+static void RemoveCommonOptionsUnwanted(OptionList& list)
+{
+    list.erase(remove_if(list.begin(),
```

Comment:
Where do these unwanted options come in from? Our "normal" non-comgr path?

Reply:
In short, yes, including the `HIP_COMPILER_FLAGS` macro defined by cmake.